tensorflow/contrib/gan/python/estimator/python/gan_estimator_impl.py

```diff
@@ -132,6 +133,8 @@ def __init__(self,
         work.
       discriminator_optimizer: Same as `generator_optimizer`, but for the
         discriminator updates.
+      get_hooks_fn: A function that takes a GANTrainOps tuple and returns a list
+        of hooks.
```

Comment:
Since most GANEstimator users won't be familiar with `get_hooks_fn`, can you describe what the list of hooks means (semantically)? Also, can you describe the default behavior.

Reply:
The list of hooks represents `training_hooks` to run during `estimator` training. The documentation is taken from `gan.train.get_sequential_train_hooks()`.
 In the current implementation it will always default to `gan.train.get_sequential_train_hooks()` which runs the generator for `n` steps and the discriminator for `m` steps.
If a user would want to run a custom operation in between the generator training and the discriminator training it is currently not possible to do so.
This change enables the user to build custom training approaches like it is already done with `gan.train.get_joint_train_hooks()` but which is not currently possible to select.

Furthermore as far as my understanding goes without these changes it is not possible for a `GANEstimator` user to specify `n` and `m` (steps to train generator & discriminator) instead it always train for `n=1` and `m=1`.

Reply:
@sleighsoft I meant in the code docstring :)

Reply:
I hope I covered everything :D